consensus/consensus-types/src/request_response.rs

```diff
@@ -1,54 +1,43 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::common::{Payload, PayloadFilter};
+use crate::{
+    common::{Payload, PayloadFilter},
+    utils::PayloadTxnsSize,
+};
 use anyhow::Result;
 use futures::channel::oneshot;
 use std::{fmt, fmt::Formatter, time::Duration};
 
+#[derive(Debug)]
+pub struct GetPayloadRequest {
+    // max txns
+    pub max_txns: PayloadTxnsSize,
+    pub max_unique_txns: u64,
+    // target txns with opt batches in max_txns as pct
+    pub opt_batch_txns_pct: u8,
+    // max inline txns
+    pub max_inline_txns: PayloadTxnsSize,
+    // return non full
+    pub return_non_full: bool,
+    // block payloads to exclude from the requested block
+    pub filter: PayloadFilter,
+    // callback to respond to
+    pub callback: oneshot::Sender<Result<GetPayloadResponse>>,
+    // block timestamp
+    pub block_timestamp: Duration,
+}
+
 pub enum GetPayloadCommand {
     /// Request to pull block to submit to consensus.
-    GetPayloadRequest(
-        // max number of transactions in the block
-        u64,
-        // max number of unique transactions in the block
-        u64,
-        // max byte size
-        u64,
-        // max number of inline transactions (transactions without a proof of store)
-        u64,
-        // max byte size of inline transactions (transactions without a proof of store)
-        u64,
-        // return non full
-        bool,
-        // block payloads to exclude from the requested block
-        PayloadFilter,
-        // callback to respond to
-        oneshot::Sender<Result<GetPayloadResponse>>,
-        // block timestamp
-        Duration,
-    ),
+    GetPayloadRequest(GetPayloadRequest),
 }
 
 impl fmt::Display for GetPayloadCommand {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         match self {
-            GetPayloadCommand::GetPayloadRequest(
-                max_txns,
-                max_txns_after_filtering,
-                max_bytes,
-                max_inline_txns,
-                max_inline_bytes,
-                return_non_full,
-                excluded,
-                _,
-                block_timestamp,
-            ) => {
-                write!(
-                    f,
-                    "GetPayloadRequest [max_txns: {}, max_txns_after_filtering: {}, max_bytes: {}, max_inline_txns: {}, max_inline_bytes:{}, return_non_full: {},  excluded: {}, block_timestamp: {:?}]",
-                    max_txns, max_txns_after_filtering, max_bytes, max_inline_txns, max_inline_bytes, return_non_full, excluded, block_timestamp
-                )
+            GetPayloadCommand::GetPayloadRequest(request) => {
+                write!(f, "{:?}", request)
```

Comment:
I'm a bit concerned about using `"{:?}"` here. That means everything recursively uses `"{:?}"` so if something like a vector was ever added to the struct, then it could explode. I think in general debug logging should be used carefully

Reply:
Fixed it.